gnuplot/src/color.rs

```diff
@@ -0,0 +1,413 @@
+			{
+				let s = s.to_string();
+				s.starts_with("0x") && s.chars().count() == 10
+					|| s.starts_with("#") && s.chars().count() == 9
+			}
+			ARGBInteger(_, _, _, _) | VariableARGBInteger(_) => true,
+			_ => false,
+		}
+	}
+}
+
+fn from_argb(a: ColorComponent, r: ColorComponent, g: ColorComponent, b: ColorComponent)
+	-> ColorInt
+{
+	((a as ColorInt) << 24) + ((r as ColorInt) << 16) + ((g as ColorInt) << 8) + (b as ColorInt)
+}
+
+fn float_color_to_int(v: f64) -> u8
+{
+	if !(0.0..=1.0).contains(&v)
+	{
+		panic!(
+			"Float value must be greater than zero and less than one. Actual value: {}",
+			v
+		);
+	}
+	((v * 255.0).round()) as u8
+}
+
+/// Converts a set of `f64` red, green and blue values in the range `0 <= x <= 1` to a 3-tuple of `u8` suitable for use as
+/// an [RGBInteger]
+///
+/// Panics if any of the arguments are not in the range `0 <= x <= 1`
+///
+/// Ses also [floats_to_argb]
+///
+/// # Arguments
+/// * r - red. 0: no red, 1: fully red
+/// * g - green. 0: no green, 1: fully green
+/// * b - blue. 0: no blue, 1: fully blue
+pub fn floats_to_rgb(r: f64, g: f64, b: f64) -> RGBInts
+{
+	(
+		float_color_to_int(r),
+		float_color_to_int(g),
+		float_color_to_int(b),
+	)
+}
+
+/// Converts a set of `f64` red, green and blue values in the range `0 <= x <= 1` to a 3-tuple of `u8` suitable for use as
+/// an [ARGBInteger]
+///
+/// Panics if any of the arguments are not in the range `0 <= x <= 1`
+///
+/// Ses also [floats_to_rgb]
+///
+/// # Arguments
+/// * a - alpha (transparency) value. 0: completely opaque, 1: completely transparent.
+/// * r - red. 0: no red, 1: fully red
+/// * g - green. 0: no green, 1: fully green
+/// * b - blue. 0: no blue, 1: fully blue
+pub fn floats_to_argb(a: f64, r: f64, g: f64, b: f64) -> ARGBInts
+{
+	(
+		float_color_to_int(a),
+		float_color_to_int(r),
+		float_color_to_int(g),
+		float_color_to_int(b),
+	)
+}
+
+impl<'l> From<&'l str> for ColorType<String>
+{
+	/// Converts `&str` into [RGBString]
+	fn from(value: &'l str) -> Self
+	{
+		ColorType::RGBString(String::from(value))
+	}
+}
+
+impl<'l> From<String> for ColorType<String>
+{
+	/// Converts `String` into [RGBString]
+	fn from(value: String) -> Self
+	{
+		ColorType::RGBString(value)
+	}
+}
+
+impl<'l> From<&'l str> for ColorType<&'l str>
+{
+	/// Converts `&str` into [RGBString]
+	fn from(value: &'l str) -> Self
+	{
+		ColorType::RGBString(value)
+	}
+}
+
+impl<T> From<ARGBInts> for ColorType<T>
+{
+	/// Converts `(u8, u8, u8, u8)` into [ARGBInteger]
+	fn from(value: ARGBInts) -> Self
+	{
+		ColorType::ARGBInteger(value.0, value.1, value.2, value.3)
+	}
+}
+
+impl<T> From<RGBInts> for ColorType<T>
+{
+	/// Converts `(u8, u8, u8)` into [RGBInteger]
+	fn from(value: RGBInts) -> Self
+	{
+		ColorType::RGBInteger(value.0, value.1, value.2)
+	}
+}
+
+impl<T> From<(f64, f64, f64)> for ColorType<T>
+{
+	/// Converts `(f64, f64, f64)` into [RGBInteger].
+	/// All values must be in the range 0-1, or the function will panic.
```

Comment:
I liked the `0 <= x <= 1` spelling of this that you used earlier.

Reply:
On this one:  Agree on the doc change, but I think on reflection, this should be a `TryFrom` and not panic (but return a `Result`. (Docs will need updating anyway)

In general, this library is relatively likely to panic (probably because it interfaces with an external program that causes fatal errors anyway), and I wanted to keep all variants on `Into`, but I now don't think that any more.  Unless you have a strong opinion I'll change fallible conversions to `TryInto`.

Reply:
Sure, that seems reasonable.